woocommerce/wc-admin
# Dev/e2e reset

Fixes #

Todo:

- [x] Reset onboarding and onboarding settings
- [x] Clear specific options (@adrianduffell - reset plugin) [Partially done](https://github.com/woocommerce/woocommerce-reset/blob/trunk/woocommerce-reset.php#L22-L33)
- [x] Reset/clear all notes (@louwie17 close to done)
- [x] Clear transients (@adrianduffell - reset plugin)
- [x] Scheduled tasks?
- [x] Clear products/orders (@louwie17 )
- [x] Deactivate/remove plugins (@louwie17 )
- [x] Clear analytics data (should be auto cleared after running action scheduler and deleting all products and orders)
- [ ] Clear user settings (ex: `woocommerce_meta.dashboard_sections`) (This can wait)

### Screenshots

### Detailed test instructions:

- Run the E2E tests locally (`npx wc-e2e docker:up && npx wc-e2e test:e2e`)
- All tests should pass
- Load the page `localhost:8084` and check if only the required plugins are still installed (woocommerce, woocommerce-admin, woocommerce-reset, basic-auth )
- Go to **WooCommerce > Home** and open your console
- Run this `wp.apiFetch({ method: 'DELETE', path: '/woocommerce-reset/v1/state'});` in your console and refresh the page
- The onboarding wizard should load again and non of the fields should be auto-filled.
